scripts/generate-compatibility-json.py

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/python3
+
+import argparse
+import json
+import re
+from typing import List, Dict
+
+import requests
+import semver
+
+COMPATIBILITY_FLAG = "compatibility"
+FROM_VERSION = "from_version"
+# FROM_VERSIONS should be specified on individual tests if the features under test are only supported
+# from specific versions of release lines.
+FROM_VERSIONS = "from_versions"
+# fields will contain arbitrary key value pairs in comments that use the compatibility flag.
+FIELDS = "fields"
+TEST_SUITE = "test_suite"
+TESTS = "tests"
+RELEASES_URL = "https://api.github.com/repos/cosmos/ibc-go/releases"
+CHAIN_A = "chain-a"
+CHAIN_B = "chain-b"
+HERMES = "hermes"
+DEFAULT_IMAGE = "ghcr.io/cosmos/ibc-go-simd"
+RLY = "rly"
+# MAX_VERSION is a version string that will be greater than any other semver version.
+MAX_VERSION = "9999.999.999"
+
+
+def parse_version(version: str) -> semver.Version:
+    """
+    parse_version takes in a version string which can be in multiple formats,
+    and converts it into a valid semver.Version which can be compared with each other.
+    The version string is a docker tag. It can be in the format of
+    - main
+    - v1.2.3
+    - 1.2.3
+    - release-v1.2.3 (a tagged release)
+    - release-v1.2.x (a release branch)
+    """
+    if version.startswith("v"):
+        # semver versions do not include a "v" prefix.
+        version = version[1:]
+    if version.startswith("release-"):
+        # strip off the release prefix and parse the actual version
+        return parse_version(version[len("release-"):])
```

Comment:
[nit] for consistency, we can either do above `return parse_version(version[1:])` or here `version = version[len("release-"):]`